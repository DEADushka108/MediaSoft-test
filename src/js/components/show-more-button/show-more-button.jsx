import React from 'react';
import PropTypes from 'prop-types';
import {ActionCreator} from '../../store/ships/ships';
import {connect} from 'react-redux';
import {shipDetails} from '../../types/ships';

const ShowMoreButton = (props) => {
  const {list, count, onClick} = props;

  return (count < list.length) ? <button className="catalog__button" type="button" onClick={onClick}>Show more</button> : null;
};

ShowMoreButton.propTypes = {
  list: PropTypes.arrayOf(shipDetails).isRequired,
  count: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onClick() {
    dispatch(ActionCreator.setCardCount());
  },
});

export {ShowMoreButton};
export default connect(null, mapDispatchToProps)(ShowMoreButton);
