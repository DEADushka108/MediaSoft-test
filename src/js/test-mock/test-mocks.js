export * from './const/const';
export * from './ships/ships';
export * from './store/store';
