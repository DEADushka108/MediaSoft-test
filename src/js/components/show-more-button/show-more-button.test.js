import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {shipsList, store, noop, MAX_CARD_COUNT} from '../../test-mock/test-mocks';
import {ShowMoreButton} from './show-more-button';

it(`Should ShowMoreButton component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <ShowMoreButton
          list={shipsList}
          count={MAX_CARD_COUNT}
          onClick={noop}
        />
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should not ShowMoreButton component render`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <ShowMoreButton
          list={shipsList}
          count={20}
          onClick={noop}
        />
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
