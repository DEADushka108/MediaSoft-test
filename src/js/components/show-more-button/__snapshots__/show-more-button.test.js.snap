// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ShowMoreButton component render correctly 1`] = `
<button
  className="catalog__button"
  onClick={[Function]}
  type="button"
>
  Show more
</button>
`;

exports[`Should not ShowMoreButton component render 1`] = `null`;
